src/main/java/io/tiledb/spark/TileDBDataReaderPartitionScan.java

```diff
@@ -217,7 +217,11 @@ public ColumnarBatch get() {
       if (sparkSchema.fields().length == 0) {
         // TODO: materialize the first dimension and count the result set size
         try (Dimension dim = domain.getDimension(0)) {
-          nRows = getDimensionColumn(dim.getName(), 0);
+          if (dim.getCellValNum() > 1)
```

Comment:
`>1`, is not a value check for variable length, you need to use the VAR_NUM check which you wrapped in `isVar`: https://github.com/TileDB-Inc/TileDB-Java/blob/8037d3f7b169229f3a83b442a7e215f838b26f1d/src/main/java/io/tiledb/java/api/Dimension.java#L297